### PR TITLE
Implement beautiful OTP input for adding phone number in registration

### PR DESCRIPTION
**Pull request by bagvand**
Steps to see new UI
1-Go to register
2-add phone number
3-see the OTP input screen.

---
This PR currently has none of the required changelog labels.

A reviewer can add one of: `T-Deprecation`, `T-Enhancement`, `T-Defect`, `T-Task` to indicate what type of change this is, or add `Type: [enhancement/defect/task]` to the description and I'll add them for you.

### CODE REVIEW

File: src/IConfigOptions.ts [hunk 1]
```diff
@@ -91,6 +91,8 @@ export interface IConfigOptions {
 
     setting_defaults?: Record<string, any>; // <SettingName, Value>
 
+    otp_length?: number;
```
Comment (t3chguy) [inline src/IConfigOptions.ts:94]: Given this OTP varies in length per Matrix homeserver and not per Element instance, this cannot be configurable here

Comment (bagvand) [reply src/IConfigOptions.ts:94]: Yes, we need OTP length from response of api `_matrix/client/r0/register/msisdn/requestToken`

Comment (t3chguy) [reply src/IConfigOptions.ts:94]: That doesn't seem to be an existing field, this change would need an MSC - https://github.com/matrix-org/matrix-spec-proposals#the-matrix-spec-process

**Comment by t3chguy** [comment]
Closing in favour of OIDC

**Pull request closed without merging**